src/common/transformations/src/transformations/transpose_sinking/ts_unary.cpp

```diff
@@ -40,41 +42,46 @@ NodePair SwapNodes(const NodePtr& first_node, const NodePtr& second_node) {
 
     auto first_node_inputs = first_node->input_values();
     first_node_inputs[0] = new_first_node;
-    auto new_second_node = first_node->clone_with_new_inputs(first_node_inputs);
 
+    auto new_second_node = first_node->clone_with_new_inputs(first_node_inputs);
     new_second_node->set_friendly_name(second_node->get_friendly_name());
-    ov::copy_runtime_info({first_node, second_node}, {new_first_node, new_second_node});
 
+    ov::copy_runtime_info({first_node, second_node}, {new_first_node, new_second_node});
     ov::replace_node(second_node, new_second_node);
 
     return std::make_pair(new_first_node, new_second_node);
 }
 
+NodePtr GetPatternNode(const PatternValueMap& pattern_to_output, const NodeVector& nodes) {
+    for (const auto& node : nodes) {
+        auto it = pattern_to_output.find(node);
+        if (it == pattern_to_output.end())
+            continue;
+        return it->second.get_node_shared_ptr();
+    }
+    return {};
+}
 }  // namespace
 
 TSUnaryForward::TSUnaryForward() {
     MATCHER_SCOPE(TSUnaryForward);
-
     auto transpose_label = wrap_type<Transpose>({any_input(), any_input()});
-    auto unary_label =
+    auto fq_label = wrap_type<FakeQuantize>({transpose_label, any_input(), any_input(), any_input(), any_input()});
+    auto unary_op_label =
         wrap_type<UnaryElementwiseArithmetic, Clamp, Elu, SoftPlus, LogicalNot, Convert, IsInf, IsNaN, IsFinite>(
             {transpose_label});
-
+    auto unary_label = std::make_shared<Or>(OutputVector{fq_label, unary_op_label});
     ov::matcher_pass_callback matcher_pass_callback = [=](Matcher& m) {
         const auto& pattern_to_output = m.get_pattern_value_map();
         auto transpose = pattern_to_output.at(transpose_label).get_node_shared_ptr();
-        auto unary = pattern_to_output.at(unary_label).get_node_shared_ptr();
-
+        auto unary = GetPatternNode(pattern_to_output, NodeVector{unary_op_label, fq_label});
```

Comment:
Will be it be faster to take Gather output node instead of searching patterns in the map?

Reply:
Could you please explain your idea? Do you want to search unary operation within input/output nodes? It seems to be the same. What do you think?